testing/profiler-tests/src/test/java/com/splunk/opentelemetry/ProfilerSmokeTest.java

```diff
@@ -74,13 +82,57 @@ static void teardown() {
   }
 
   @Test
-  void ensureJfrFilesCreated() {
+  void ensureJfrFilesContainContextChangeEvents() throws Exception {
     System.out.println("Petclinic has been started.");
 
+    generateSomeSpans();
+
     await()
         .atMost(60, TimeUnit.SECONDS)
         .pollInterval(1, TimeUnit.SECONDS)
-        .untilAsserted(() -> assertThat(findJfrFilesInOutputDir()).isNotEmpty());
+        .untilAsserted(() -> assertThat(spanThreadContextEventsFound()).isTrue());
+  }
+
+  private boolean spanThreadContextEventsFound() throws Exception {
+    List<Path> files = findJfrFilesInOutputDir();
+    return files.stream().anyMatch(this::containsContextAttached);
+  }
+
+  private boolean containsContextAttached(Path path) {
+    try (RecordingFile file = new RecordingFile(path)) {
+      while (file.hasMoreEvents()) {
+        RecordedEvent event = file.readEvent();
+        if ("otel.ContextAttached".equals(event.getEventType().getName())) {
+          return true;
+        }
+      }
+    } catch (IOException e) {
+      return false;
+    }
+    return false;
+  }
+
+  private void generateSomeSpans() throws Exception {
+    System.out.println("Generating some spans...");
```

Comment:
probably don't want printlns in here. log, maybe?

Reply:
I had hoped to see them in the build stdout but I think gradle prevents that from actually working.  Does using a logger help with that?  I just want some indication of how far the thing progressed before the monolithic pass/fail happens.  Open to ideas.

Reply:
should matter logger or no. You have to run at --info to see either, correct?

Reply:
🤷🏻 I dunno, probably.  I'm flexible, so if we prefer logs then I can change it.

Reply:
I do prefer using a logger here, so if we can get that to work, then 👍🏼 . Otherwise, whatever gets things up and running if fine with me in test code. (do we expect to actually need this output, or was it just useful when getting things running initially?)

Reply:
It's super helpful if/when things break. Honestly, without the output it's just an opaque "test failed, good luck buddy".  At least with some output you can get an indication of how far it got before it failed.